### PR TITLE
[sc-106495] cleartext password

### PR DESCRIPTION
**Pull request by alexbourret**
Approach followed in this POC:

- Priority on keeping existing flows working
- Replace the 'Use preset' boolean with an auth method selector

Implementation details / consequences:

- The 'Use preset' boolean is hidden. It is still processed if present in the flow's json but cannot be add on new sync recipes.
- A new selector is added, with no default. If absent, the old way of selecting the active preset is used (using the Use preset info). If any auth method is selected, it takes absolute priority in the display of preset info.
- The preset displayed in the interface is only the one selected in the preset (nothing displayed if selector left untouched).
    - The behavior will look incoherent at first (recipe still working but presets invisible). The user will have to select a preset to see the one(s) already stored in the recipe, and is forced to make a choice.

### CODE REVIEW

**Comment by alexbourret** [comment]
Improved solution: the previous settings (if exists) is displayed when the `Authentication type` selector is left on `Nothing selected`.

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]
A small nitpick: rename `basic-oauth`.

Tested:
- Export with the new per user credential: OK

Plugin Update with the following export to folder:
- inline settings
- a manual preset
- a instance-wide preset defined in the admin
- a project-wide preset 

Test case:
- Plugin update:
  - inline settings are still present
  - manual preset setting are stil present
  - previously selected preset present 
- Authentication type is not selected
- Run of a scenario building all the existing exports as is 
- Save with auth type = None
    - Manual preset 
    - Preset
    - Settings
- Save auth type
    - Manuel Preset → Preset legacy
    - Preset → Preset legacy
    - Settings → user/password legacy

File: python-exporters/tableau-hyper_upload/exporter.json [hunk 1]
```diff
@@ -17,40 +17,67 @@
             "label":"Tableau Server",
             "type": "SEPARATOR"
         },
+        {
+            "name": "auth_type",
+            "label": "Authentication type",
+            "type": "SELECT",
+            "selectChoices": [
+                {
+                    "value": "basic-oauth",
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-exporters/tableau-hyper_upload/exporter.json]: Done in [a84dcba](https://github.com/dataiku/dss-plugin-tableau-hyper/pull/17/commits/a84dcba37d128c580d0d4ca81a348545cbac08a7)

**Review by MayeulRousselet — APPROVED** [review]
LGTM

**Pull request merged**